src/Kestrel.Core/Internal/Http/HttpParser.cs

```diff
@@ -279,16 +280,20 @@ public unsafe bool ParseHeaders(TRequestHandler handler, ReadOnlyBuffer buffer,
                             else
                             {
                                 var current = reader.Position;
+                                var currentSlice = buffer.Slice(current, bufferEnd);
```

Comment:
@pakrym Won't this performance? This is why I didn't understand why we were removing seek that just took cursors. This means we end up slicing more than we need to.

Reply:
Hopefully we don’t regress performance any more than we already have